datahub/main.py

```diff
@@ -93,13 +94,15 @@ def create_opal_data(data: OpalDictData | OpalArrayData) -> dict[str, str]:
     return {"message": "Data submitted successfully."}
 
 
+# TODO: Fix return typing annotation
 @app.get("/opal")
-def get_opal_data() -> dict[str, float]:
+def get_opal_data() -> dict[str | Hashable, Any]:  # type: ignore[misc]
     """GET method function for getting Opal Dataframe as JSON.
 
     Returns:
         A Dict of the Opal Dataframe in JSON format
     """
-    # Requires opal-test branch to be merged
-    # return dt.opal_df.to_json()
-    return {"test": 0}
+    data = dt.opal_df.to_dict()
```

Comment:
Keys in json must be strings, so the index values are converted into strings when this is used. We can represent the DF in a different orientation. I think "split" orientation works best - that matches the key-value pairs that will be input in the the `pd.DataFrame` constructor ("index", "columns" and "data").
```suggestion
    data = dt.opal_df.to_dict(orient="split")
```

See here https://stackoverflow.com/a/44127109